### ISSUE TITLE
Tags are a pain in new Azure.Core based .Net SDK modules

### ISSUE THREAD

**Issue by asmithgt** [Azure/azure-sdk-for-net#18378]
Managing Tags on Generic Resources is VERY FRUSTRATING and basically impossible with the new Azure.Core modules.  For one the GenericResources type from the Microsoft.Azure.Management.Fluent module can not update Tags.  It doens't have the logic.  Secondly, it is also a Read only dictionary which is also very frustrating.

So fine, I'll use the ResoucresManagementClient. Specifically the TagsOperations component.  But that doesn't work either because it seems like there is nothing I can do to create a "Tags" object type.

In summary... I'm horribly frustrated.  Why is this so difficult?  Can it not be a little easier?  Can the new Azure.Core based modules be a little more workable?  Maybe some better examples in the documentation? Please?

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.

**Comment by m-nash** [comment]
This is something we are addressing in our upcoming preview release of the management plane SDK.  We will be providing helper methods to Add, Remove, or Set the tags without needing to deal with the read only dictionary.

In the mean time can you provide a code sample on where you are trying to update the tags and I can help provide a work around in the meantime.

**Comment by asmithgt** [comment]
@m-nash Thank you for your response and I'm very glad to hear that this is being addressed.  After I created this issue, I continued to bang my head on the wall until I finally came up with a solution involving a combination of the IGenericResources interface off of Microsoft.Azure.Management.Fluent along with the ResourceManagementClient from Microsoft.Azure.Management.ResourceManager.  It's a bit cludgey, but here is a snipped of the code.

```
var rmClient = new ResourceManagementClient(myRestClient);

rmClient.SubscriptionId = subscriptionId;

var rDetails = GetReourceDetails(rmClient, resourceId, log);

var myResourceGeneric = azure.GenericResources.GetById(resourceId);

if (!myResourceGeneric.Tags.ContainsKey("CreatedBy")) {
            myResourceGeneric.Inner.Tags["CreatedBy"] = user;
}
if (!myResourceGeneric.Tags.ContainsKey("CreatedDate")) {
     myResourceGeneric.Inner.Tags["CreatedDate"] = time.ToString("u");
}
myResourceGeneric.Inner.Tags["ModifiedBy"] = user;
myResourceGeneric.Inner.Tags["ModifiedDate"] = time.ToString("u");

var asyncResult = rmClient.Resources.UpdateByIdAsync(resourceId,rDetails.ApiVersion, myResourceGeneric.Inner); 
```
It may be easier to accomplish this in some other way, but it was pretty frustrating that .Net SDK documentation for Authentication is setup for using the latest preview releases of the SDK, but I have to go back and find different clients that authenticate in a little different manner and throw them together when I would hope to be able to Get and Set with a single client class using the documented authentication mechanisms.

Thanks again for the feedback and I look forward to working with new and improved versions of the .Net SDK for Azure.
